DOC: Patch new flake8 command grep

gfyoung opened the pull request:
The grep was initially matching to "pandas," which is incorrect because that was also matching files containing "pandas" in the name but that were not in the main `pandas` directory (e.g. performance test code).  This change enforces that we match to any Python files in the main `pandas` directory.

Also picked up compatibility issue with OSX, in which the `-r` flag does not exist.  However, `xargs` terminates if the argument list is empty, which was the whole point of passing in `-r` in the first place.

Follow-up to #15712

gfyoung committed:
DOC: Patch new flake8 command grep

The grep was initially matching to "pandas,"
which is incorrect because that was also
matching files containing "pandas" in the
name but that were not in the main pandas
directory (e.g. performance test code)

Follow-up to gh-15712

[ci skip]

jreback commented on doc/source/contributing.rst:
```diff
@@ -527,7 +527,7 @@ unused function. However, style-checking the diff will not catch this because
 run this command, though it will take longer::
 
-   git diff master --name-only -- '*.py' | grep 'pandas' | xargs -r flake8
+   git diff master --name-only -- '*.py' | grep 'pandas/' | xargs -r flake8
```
not sure if its worth noting but need to remove the ``-r`` on osx :<

otherwise this command looks good to me (I use it!)

gfyoung replied:
Awesome!  But how do you tell `xargs` then to abort the command if the list is empty on OSX?

jreback replied:
it just works
```
(pandas) bash-3.2$ git diff master --name-only -- '*.py' | grep 'pandas/' | xargs -r flake8
xargs: illegal option -- r
usage: xargs [-0opt] [-E eofstr] [-I replstr [-R replacements]] [-J replstr]
             [-L number] [-n number [-x]] [-P maxprocs] [-s size]
             [utility [argument ...]]
(pandas) bash-3.2$ git diff master --name-only -- '*.py' | grep 'pandas/' | xargs flake8
```

gfyoung replied:
Hmm...I can't seem to find a `-r` flag for OSX.  Annoying...but I can change the documentation to state that this could be an issue.

gfyoung replied:
Oh, interesting.  I did not realize, it just aborts like that.  Okay, then I'll mention that in the docs.

gfyoung replied:
@jreback : Made doc patch regarding OSX compatibility with `xargs`

jreback replied:
what does ``-r`` do anyhow?

gfyoung replied:
On Linux, it means "abort immediately if list is empty"

jreback replied:
oh I c, ok then.

gfyoung committed:
COMPAT: Do not run xargs with -r on OSX

OSX xargs has no -r, but it still terminates
when xargs is empty, which was the whole point
of passing in that flag in the first place.

[ci skip]

jreback commented:
thanks!
